### ISSUE TITLE
Default LR scheduling - steps vs. epochs

### ISSUE THREAD

**Issue by rbommasanimozilla** [OpenNMT/OpenNMT-py#822]
With the new change to steps, I believe the current behavior is to validate once every 10k steps. This to me seems reasonable. However, it seems that the LR scheduling treats this 10k steps as the point to decay the LR. This in practice seems to be too fast and seems to be pretty far from the previous decaying based on number of epochs (of course depending on dataset size).

**Comment by vince62s** [comment]
Default is:
After 50000 steps start to decay and decay LR each 10000 steps after.
Feel free to change these values anyway.

**Comment by rbommasanimozilla** [comment]
Right, i am just wondering if these are good defaults

**Comment by livenletdie** [comment]
I am not sure if this issue is related to the behavior I am seeing or not but I updated to the latest version of OpenNMT-py and re-ran the transformer baseline training for En De dataset. The validation perplexity I am seeing are much higher (worse) than with the older version of the repository. The command I am using is the same one from Bug 637:

python train.py -gpuid 0 -rnn_size 512 -word_vec_size 512 -batch_type tokens -save_model ./checkpointModels/transformerBug637/baseline \
    -batch_size 4096 -accum_count 4 -train_steps 500000  -max_generator_batches 32 \
    -normalization tokens -dropout 0.1 -max_grad_norm 0 -optim adam \
    -encoder_type transformer -decoder_type transformer -position_encoding \
    -param_init 0 -warmup_steps 16000 -learning_rate 2 -save_checkpoint_steps 10000 -keep_checkpoint 5 \
    -decay_method noam -label_smoothing 0.1 -adam_beta2 0.998 \
    -data ../datasets/enDe/processed -param_init_glorot -layers 6 -report_every 1000 

My validation perplexity hovers around 4.9 after 280K steps while it as closer 4.2 in the original case.

**Comment by vince62s** [comment]
it is not related at all;
But what I can tell is that onthe DE to ENtask I have erified I retrieve the same results.
Didn't you change the preprocess or such ?

**Comment by livenletdie** [comment]
Thank you for the comment and good to know. Let me double check my raw dataset pointers to see if something changed. I used the same preprocess command from the bug

python preprocess.py -train_src ../datasets/enDe/train.en \
    -train_tgt ../datasets/enDe/train.de \
    -valid_src ../datasets/enDe/valid.en -valid_tgt ../datasets/enDe/valid.de \
    -save_data ../datasets/enDe/processed \
    -src_seq_length 100 -tgt_seq_length 100 \
    -max_shard_size 200000000 -share_vocab

**Comment by vince62s** [comment]
closing this, open a new issue if you need more help.
